Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/sharing/FileSharing.kt

```diff
@@ -0,0 +1,49 @@
+package de.rki.coronawarnapp.util.sharing
+
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import androidx.core.content.FileProvider
+import dagger.Reusable
+import de.rki.coronawarnapp.BuildConfig
+import de.rki.coronawarnapp.util.di.AppContext
+import timber.log.Timber
+import java.io.File
+import javax.inject.Inject
+
+@Reusable
+class FileSharing @Inject constructor(
+    @AppContext private val context: Context
+) {
+
+    private fun getFileUri(path: File): Uri = FileProvider.getUriForFile(
+        context,
+        AUTHORITY,
+        path
+    )
+
+    fun getIntent(
+        path: File,
+        title: String,
+        body: String? = null
+    ): Intent = Intent(Intent.ACTION_SEND).apply {
+        type = determineMimeType(path)
+        addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
```

Comment:
In the contact journal, we used the `ShareCompat.IntentBuilder` instead. See `ContactDiaryOverviewFragment.exportLocationsAndPersons`